File: 백준/Silver/11501. 주식/주식.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
main();

function main(){
    let answer ="";
    for(let i = 2; i < input.length; i += 2) {
        answer += calc(i) + "\n";
    }
    console.log(answer);
}

function calc(i){
    let stockFlow = input[i].trim().split(' ').map(e => parseInt(e));
    let profit = 0;
    let maxPrice = 0;
    for(let j = stockFlow.length - 1; j >= 0; j--) {
        if (stockFlow[j] >= maxPrice) {
            maxPrice = stockFlow[j];
        }
        profit += maxPrice - stockFlow[j];
    }
    return profit;
}

File: 백준/Silver/11501. 주식/주식.py
import sys
input = sys.stdin.readline

t = int(input())
answer = []
for _ in range(t):
  n = int(input())
  prices = list(map(int,input().split(' ')))
  maxPrice = 0
  total = 0
  
  for i in range(n-1,-1,-1):
    if maxPrice < prices[i]:
      maxPrice = prices[i]
    else:
      total += maxPrice-prices[i]
  answer.append(str(total))

print('\n'.join(answer))